### PR TITLE
feat: support for using apisix 3.x image

### LINKED ISSUE #389: bug: using the latest image(3.0.0-centos) to upgrade helm release failed: container cannot connect to etcd
### Current Behavior

When i upgraded the image of helm release from 2.15.0-centos(or 2.15.0-alpine) to the latest one, the latest container failed to run due to failing to connect to etcd cluster. The error message in container logs like:

```bash
/usr/local/openresty/luajit/bin/luajit ./apisix/cli/apisix.lua init
/usr/local/openresty/luajit/bin/luajit ./apisix/cli/apisix.lua init_etcd
Warning! Request etcd endpoint 'http://127.0.0.1:2379/version' error, connection refused, retry time=1
Warning! Request etcd endpoint 'http://127.0.0.1:2379/version' error, connection refused, retry time=2
request etcd endpoint 'http://127.0.0.1:2379/version' error, connection refused
all etcd nodes are unavailable
```

I have recorded all the steps to reproduce this bug using asciinema, see: https://asciinema.org/a/4q10QLD1kJw3N5viGAR7JgYd3

### Expected Behavior

Upgraded successfully.

### Error Logs

/usr/local/openresty/luajit/bin/luajit ./apisix/cli/apisix.lua init
/usr/local/openresty/luajit/bin/luajit ./apisix/cli/apisix.lua init_etcd
Warning! Request etcd endpoint 'http://127.0.0.1:2379/version' error, connection refused, retry time=1
Warning! Request etcd endpoint 'http://127.0.0.1:2379/version' error, connection refused, retry time=2
request etcd endpoint 'http://127.0.0.1:2379/version' error, connection refused
all etcd nodes are unavailable

### Steps to Reproduce

https://asciinema.org/a/4q10QLD1kJw3N5viGAR7JgYd3

### Environment

- APISIX version (run `apisix version`): 3.0.0
- Operating system (run `uname -a`):
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

### PR DESCRIPTION
**Pull request by tzssangglass**
Fix #389

### CODE REVIEW

File: charts/apisix/templates/configmap.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-{{- $isBeta := (semverCompare ">=2.99.0" .Values.apisix.image.tag) }}
+{{- $isBeta := (semverCompare ">=2.99.0-debian" .Values.apisix.image.tag) }}
```
Comment (tzssangglass) [inline charts/apisix/templates/configmap.yaml:17]: `>=2.99.0-debian` is a bad syntax, I will try to update it.

File: docs/en/latest/apisix.md [hunk 2]
```diff
@@ -31,6 +31,18 @@ helm repo update
...
+
+The current default version of APISIX installed is 2.15.0, if you want to install APISIX 3.x, to install the chart of APISIX with:
+
+```shell
+helm install apisix apisix/apisix --set apisix.image.tag=3.0.0-debian --create-namespace --namespace apisix
+```
+
+:::note
+
+apisix-ingress-controller is not compatible with APISIX 3.x.
```
Comment (tao12345666333) [inline docs/en/latest/apisix.md:44]: the dashboard is also not compatible with APISIX 3.x, right?

**Comment by tzssangglass** [comment]
- 2.99.0-debian
- 2.99.0-centos
- 3.0.0-debian
- 3.0.0-centos

work around with `semverCompare ">=2.99.0-0"`, and would make `isBeta` as true and use config format of v3.

File: docs/en/latest/apisix.md (hunk 2 above)
Comment (tzssangglass) [reply docs/en/latest/apisix.md]: updated

File: docs/en/latest/apisix.md [hunk 3]
```diff
@@ -31,6 +31,18 @@ helm repo update
 helm install apisix apisix/apisix --create-namespace  --namespace apisix
 ```
 
+### APISIX v3.x
+
+The current default version of APISIX installed is 2.15.0, if you want to install APISIX 3.x, to install the chart of APISIX with:
```
Comment (tokers) [inline docs/en/latest/apisix.md:36]:
```suggestion
The current default version of APISIX installed is `2.15.0`, if you want to install APISIX 3.x, to install the chart of APISIX with:
```

File: docs/en/latest/apisix.md [hunk 4]
```diff
@@ -31,6 +31,18 @@ helm repo update
...
+
+```shell
+helm install apisix apisix/apisix --set apisix.image.tag=3.0.0-debian --create-namespace --namespace apisix
+```
+
+:::note
+
+APISIX-Ingress-Controller and APISIX Dashboard are not compatible with APISIX 3.x.
```
Comment (tokers) [inline docs/en/latest/apisix.md:44]:
```suggestion
The latest APISIX Ingress Controller and APISIX Dashboard are not compatible with APISIX 3.x.
```

File: docs/en/latest/apisix.md (hunk 3 above)
Comment (tzssangglass) [reply docs/en/latest/apisix.md]: done

File: docs/en/latest/apisix.md (hunk 4 above)
Comment (tzssangglass) [reply docs/en/latest/apisix.md]: done

**Comment by tzssangglass** [comment]
note that the `appVersion` in Chart.yaml is `2.15.0`

**Review by tokers — APPROVED** [review]

**Review by tao12345666333 — APPROVED** [review]

**Pull request merged**